### PR TITLE
GUVNOR-2761: ConditionPopup refactored and operator preserved if no f…

### PR DESCRIPTION
**Pull request by jomarko**
…ield change

### CODE REVIEW

**Comment by manstis** [comment]
Hi @karreiro since you're looking at this code for the Wizard changes could you please review? 

@jomarko Please note @karreiro is rewriting the "New Column" related UI components to use a Wizard. The existing screens may be re-usable short-term for "editing" columns; but ideally they'll all go for 7.x. Please bare this in mind; is your recent efforts - if part of a bigger strategy - could become wasted.

**Comment by karreiro** [comment]
Hi @jomarko 

Thanks for your contribution :-)

Since you moved the logic from `ConditionPopup.java` to `ConditionPopupView.java`, it's important to cover it with tests. Even being a View, there's much logic there that need to be tested.

However, as @manstis said, the existing screens will all go soon, so I think we have two options:

- I) cover the `ConditionPopupView.java` with tests;
- II) rollback the refactoring and change / cover with tests only the part that you change.

Thanks again @jomarko

**Comment by manstis** [comment]
@hasys What's the status of this PR?

**Comment by jomarko** [comment]
@manstis Still the same. I need to make decision if to add tests and merge or close, due to @karreiro 's refactoring.

**Comment by jomarko** [comment]
@karreiro would you mind have a look now?

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/ConditionPopupView.java [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+                if ( !this.presenter.isReadOnly() ) {
+                    formula.addClickHandler( new ClickHandler() {
+                        public void onClick( ClickEvent w ) {
+                            presenter.setFactField( null );
+                            presenter.applyConsTypeChange( BaseSingleFieldConstraint.TYPE_RET_VALUE );
+                        }
+                    } );
+                }
+
+                if ( !this.presenter.isReadOnly() ) {
```
Comment (karreiro) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/ConditionPopupView.java]: IMO, I would keep it simple with only one `if ( !this.presenter.isReadOnly() )`. But, it's just a personal option :-)

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/ConditionPopupTest.java [hunk 2]
```diff
@@ -67,11 +78,32 @@ public static void setupApplicationPreferences() {
...
+
+        setUpPopup( model,
+                    pattern,
+                    column,
+                    false,
+                    false );
+
+    }
+
+    private void setUpPopup(final GuidedDecisionTable52 model,
```
Comment (karreiro) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/ConditionPopupTest.java]: Formatting issue.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/ConditionPopupTest.java [hunk 3]
```diff
@@ -168,4 +172,110 @@ public void fieldIsPreSelectedWhenBeingEdited() {
...
+
+        verify( fieldChangePopUp ).hide();
+        verify( popup.view ).enableFooter( true );
+    }
+
+    @Test
+    public void testCancelFieldChangePopUp() throws Exception {
+        FormStylePopup fieldChangePopUp = mock( FormStylePopup.class );
+
+        popup.view = spy( new ConditionPopupView( popup )) ;
```
Comment (karreiro) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/ConditionPopupTest.java]: Formatting issue.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/ConditionPopupView.java (hunk 1 above)
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/ConditionPopupView.java]: Sure, It definitely makes sense.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/ConditionPopupTest.java (hunk 2 above)
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/ConditionPopupTest.java]: Sure, I will fix it.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/ConditionPopupTest.java (hunk 3 above)
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/ConditionPopupTest.java]: Sure, I will fix it.

**Comment by jomarko** [comment]
@karreiro all changes applied. Let me know if you find something more to fix.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/ConditionPopupTest.java [hunk 4]
```diff
@@ -168,4 +172,110 @@ public void fieldIsPreSelectedWhenBeingEdited() {
                       indexArgumentCaptor.getValue().intValue() );
     }
 
+    @Test
+    public void testConfirmFieldChangePopUp() throws Exception {
+        FormStylePopup fieldChangePopUp = mock( FormStylePopup.class );
+
+        popup.view = spy( new ConditionPopupView( popup )) ;
```
Comment (karreiro) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/ConditionPopupTest.java]: @jomarko Formatting issue here, too.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/ConditionPopupTest.java [hunk 5]
```diff
@@ -168,4 +172,110 @@ public void fieldIsPreSelectedWhenBeingEdited() {
...
+
+        verify( fieldChangePopUp ).hide();
+        verify( popup.view ).enableFooter( true );
+    }
+
+    @Test
+    public void testConfirmFieldChangePopUpNoChange() throws Exception {
+        FormStylePopup fieldChangePopUp = mock( FormStylePopup.class );
+
+        popup.view = spy( new ConditionPopupView( popup )) ;
```
Comment (karreiro) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/ConditionPopupTest.java]: @jomarko Formatting issue.

**Comment by jomarko** [comment]
@karreiro I fixed two more formatting issues you pointed out. Let's wait for jenkins result.

**Pull request merged**